app/components/address-autofill.tsx

```diff
@@ -0,0 +1,8 @@
+import dynamic from "next/dynamic";
```

Comment:
this file allows NextJS to load the third party lib `@mapbox/search-js-react` dynamically, and therefore will not execute it on the server side (which causes the "`document` not found" error

Reply:
More detail here: https://nextjs.org/docs/pages/building-your-application/optimizing/lazy-loading#with-no-ssr